fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/PlanHashable.java

```diff
@@ -244,6 +244,9 @@ static int objectPlanHash(@Nonnull final PlanHashMode mode, @Nullable final Obje
         if (mode.getKind() != PlanHashKind.LEGACY && obj instanceof Map) {
             return mapsPlanHash(mode, (Map<?, ?>)obj);
         }
+        if (obj instanceof PlanHashable) {
+            return ((PlanHashable)obj).planHash(mode);
+        }
```

Comment:
Changing the order here is necessary, because now a `Value` is actually an `Iterable`, therefore, it will match line 250.

If we do not change the order of lines, the following will happen:
1. a `Value` such as `RCV(x,y,z)` will match at line 250, calls into `iterablePlanHash`.
2. the method `iterablePlanHash` will iterate over the passed iterable object
3. the _first_ item in the `RCV(x,y,z)` is the `RCV` itself (because of pre-order and the recursive nature of the non-linear structure `TreeLike<>`.
4. it calls now in `objectPlanHash`
5. it goes back to 1 (!)

-> infinite recursion.

Reply:
I am assuming the plan hash has enough test coverage such that if the PRB is green (which is), then plan hash should not have been changed.

Reply:
Restored the code now that `TreeLike` does not extend `Iterable`.